include/foxy/detail/tunnel.hpp

```diff
@@ -249,13 +249,26 @@ tunnel_op<TunnelHandler>::operator()(boost::system::error_code ec,
 
         BOOST_ASIO_CORO_YIELD
         {
-          auto target =
+          auto const path =
```

Comment:
You should extract code that's not parametrized on the handler into a separate function template, to avoid bloat in the binary.